### PR TITLE
[setup] Oomph setup model + m2e mapping

### PR DESCRIPTION
**Pull request by dhuebner**
Using oomph makes it easier for contributors to get started with Che -
Development

### CODE REVIEW

**Comment by TylerJewell** [comment]
Can we place the files into another directory location - in the /assembly directory?

We are working to keep the main repository directory clean with only:
1. Markdown files for key user / developer information
2. The primary maven pom.xml for building the project.
3. The Dockerfile for building and running the file.

It's not universally intuitive what the two files names provide - so want to have them placed into a location where it's clear that they are offering Oomph configuration + appropriate README files on how it will be maintained along with the purpose of each file.

**Comment by dhuebner** [comment]
Moved to separate folder. Feel free to change anything you want.
When the setup is merged, I will ask the oomph guys to add it to the official eclipse projects catalogue.

File: assembly/oomph/lifecycle-mapping-metadata.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      </pluginExecutionFilter>
+      <action>
+        <ignore />
+      </action>
+    </pluginExecution>
+    <pluginExecution>
+      <pluginExecutionFilter>
+        <groupId>org.eclipse.che.core</groupId>
+        <artifactId>che-core-api-dto-maven-plugin</artifactId>
+        <versionRange>4.0.0-RC14-SNAPSHOT</versionRange>
```
Comment (skabashnyuk) [inline assembly/oomph/lifecycle-mapping-metadata.xml]: static version is not good?

Comment (benoitf) [inline assembly/oomph/lifecycle-mapping-metadata.xml]: we may use <version>[0.0.0,)</version> for other plugin versions as well

**Comment by skabashnyuk** [comment]
@dhuebner  How Oomph related to che? How it can helps to make it easier for contributors to get started with Che ? 
Can you describe overall picture?
Can you describe what did you do in this PR to make it happen?

**Comment by dhuebner** [comment]
@skabashnyuk Che relates to eclipse and oomph is the official eclipse installer. There are already a lot of eclipse projects that provides setup models to the Oomph catalogue.

The scope of the oomph project: 
The Oomph project provides tools that automate the provisioning and management of project-specific IDEs and workspaces, as well as other tediously repetitive tasks.

Read more here: https://projects.eclipse.org/proposals/oomph

The Che.setup file is currently doing the following for the eclipse user:
1. Sets default java to 1.8 and some more eclipse preferences
2. Installs m2e integration + Test NG plugins
3. clones the che git repository from github (stream: master)
4. Creates some working sets (che-parent, che-core, che-plugins ..etc.)
5. Runs "m2e project import" starting with the pom.xml in git root
6. Sets  lifecycle-mapping-metadata.xml as default m2e lifecycle mapping

**Comment by dhuebner** [comment]
@skabashnyuk Here is a guide for Xtext: https://github.com/eclipse/xtext/blob/master/README.md#how-to-contribute

**Comment by TylerJewell** [comment]
lgtm - i think we will put all installer + configurators in /assembly.

File: assembly/oomph/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+## About Eclipse Che
+Eclipse Che is a next generation Eclipse IDE and open source alternative to IntelliJ. This repository is licensed under the Eclipse Public License 1.0. Visit [Eclipse Che's Web site](http://eclipse.org/che) for feature information or the main [Che assembly repository](http://github.com/codenvy/che) for a description of all participating repositories.
```
Comment (benoitf) [inline assembly/oomph/README.md]: repository is github.com/eclipse/che

File: assembly/oomph/lifecycle-mapping-metadata.xml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        </goals>
+      </pluginExecutionFilter>
+      <action>
+        <configurator>
+          <id>org.eclipse.m2e.jdt.javaConfigurator</id>
+        </configurator>
+      </action>
+    </pluginExecution>
+  </pluginExecutions>
+</lifecycleMappingMetadata>
```
Comment (benoitf) [inline assembly/oomph/lifecycle-mapping-metadata.xml]: missing newline

**Comment by TylerJewell** [comment]
@dhuebner @skabashnyuk @benoitf --- update?  Are we ready to merge?

**Comment by benoitf** [comment]
It seems it needs to be rebased at some points as there are conflicts else looks OK for me

**Comment by skabashnyuk** [comment]
@dhuebner can you update this PR or create new one? It's looks a bit outdated. Better new one.

**Comment by dhuebner** [comment]
@skabashnyuk Sure, I will create a new one soon.

**Pull request closed without merging**